src/Functions/bitShiftRight.cpp

```diff
@@ -29,6 +31,51 @@ struct BitShiftRightImpl
             return static_cast<Result>(a) >> static_cast<Result>(b);
     }
 
+    static inline NO_SANITIZE_UNDEFINED void apply(const UInt8 * pos [[maybe_unused]], const UInt8 * end [[maybe_unused]], const B & b [[maybe_unused]], ColumnString::Chars & out_vec, ColumnString::Offsets & out_offsets)
+    {
+        if constexpr (is_big_int_v<B>)
+            throw Exception("BitShiftRight is not implemented for big integers as second argument", ErrorCodes::NOT_IMPLEMENTED);
+        else
+        {
+            UInt8 word_size = 8;
+            if (b >= static_cast<B>((end - pos) * word_size) || b < 0)
+            {
+                // insert default value
+                out_vec.push_back(0);
+                out_offsets.push_back(out_offsets.back() + 1);
+                return;
+            }
+
+            size_t shift_right_bytes = b / word_size;
+            size_t shift_right_bits = b % word_size;
+
+            const UInt8 * begin = pos;
+            const UInt8 * shift_right_end = end - shift_right_bytes;
+
+            const size_t old_size = out_vec.size();
+            size_t length = shift_right_end - begin;
+            const size_t new_size = old_size + length + 1;
+            out_vec.resize(new_size);
+
+            UInt8 * op_pointer = const_cast<UInt8 *>(shift_right_end);
+            out_vec[old_size + length] = 0;
+            UInt8 * out = out_vec.data() + old_size + length;
+            while (op_pointer > begin)
+            {
+                op_pointer--;
+                out--;
+                UInt8 temp_value = *op_pointer >> shift_right_bits;
+                if (op_pointer - 1 >= begin)
+                {
+                    *out = UInt8(UInt8(*(op_pointer - 1) << (8 - shift_right_bits)) | temp_value);
+                }
+                else
+                    *out = temp_value;
```

Comment:
Please, add a comment, that here we combining bits from current and previous bytes.